### PR TITLE
Added the links to the headers

### PR DESCRIPTION
**Pull request by gurkantuna**
Added the links to the headers

## Summary

I added the following URL addresses to the document for more detailed information.

  - [Native sized integers](https://github.com/dotnet/csharplang/blob/main/proposals/csharp-9.0/native-integers.md)
  - [Function pointers](https://github.com/dotnet/csharplang/blob/main/proposals/csharp-9.0/function-pointers.md)
  - [Suppress emitting localsinit flag](https://github.com/dotnet/csharplang/blob/main/proposals/csharp-9.0/skip-localsinit.md)
  - [Target-typed `new` expressions](https://github.com/dotnet/csharplang/blob/main/proposals/csharp-9.0/target-typed-new.md)
  - [`static` anonymous functions](https://github.com/dotnet/csharplang/blob/main/proposals/csharp-9.0/static-anonymous-functions.md)
  - [Target-typed conditional expressions](https://github.com/dotnet/csharplang/blob/main/proposals/csharp-9.0/target-typed-conditional-expression.md)
  - [Covariant return types](https://github.com/dotnet/csharplang/blob/main/proposals/csharp-9.0/covariant-returns.md)
  - [Extension `GetEnumerator` support for `foreach` loops](https://github.com/dotnet/csharplang/blob/main/proposals/csharp-9.0/extension-getenumerator.md)
  - [Lambda discard parameters](https://github.com/dotnet/csharplang/blob/main/proposals/csharp-9.0/lambda-discard-parameters.md)
  - [Attributes on local functions](https://github.com/dotnet/csharplang/blob/main/proposals/csharp-9.0/local-function-attributes.md)
  - [Module initializers](https://github.com/dotnet/csharplang/blob/main/proposals/csharp-9.0/module-initializers.md)
  - [New features for partial methods](https://github.com/dotnet/csharplang/blob/main/proposals/csharp-9.0/extending-partial-methods.md)

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/dotnet/docs?pullRequest=23787) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/docs?pullRequest=23787) before we can accept your contribution.<br/><br/>:x: g-u-r-k-a-n [sign now](https://cla.dotnetfoundation.org/dotnet/docs?pullRequest=23787)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/docs?pullRequest=23787) it.</sub>

File: docs/csharp/whats-new/csharp-9.md [hunk 1]
```diff
@@ -12,20 +12,20 @@ C# 9.0 adds the following features and enhancements to the C# language:
 - [Top-level statements](#top-level-statements)
 - [Pattern matching enhancements](#pattern-matching-enhancements)
 - [Performance and interop](#performance-and-interop)
-  - Native sized integers
-  - Function pointers
-  - Suppress emitting localsinit flag
+  - [Native sized integers](../../../../../../csharplang/blob/main/proposals/csharp-9.0/native-integers.md)
```
Comment (Youssef1313) [inline docs/csharp/whats-new/csharp-9.md:15]:
```suggestion
  - [Native sized integers](~/_csharplang/proposals/csharp-9.0/native-integers.md)
```

File: docs/csharp/whats-new/csharp-9.md [hunk 2]
```diff
@@ -12,20 +12,20 @@ C# 9.0 adds the following features and enhancements to the C# language:
...
+  - [Function pointers](../../../../../../csharplang/blob/main/proposals/csharp-9.0/function-pointers.md)
```
Comment (Youssef1313) [inline docs/csharp/whats-new/csharp-9.md:16]:
```suggestion
  - [Function pointers](~/_csharplang/proposals/csharp-9.0/function-pointers.md)
```

File: docs/csharp/whats-new/csharp-9.md [hunk 3]
```diff
@@ -12,20 +12,20 @@ C# 9.0 adds the following features and enhancements to the C# language:
...
+  - [Suppress emitting localsinit flag](../../../../../../csharplang/blob/main/proposals/csharp-9.0/skip-localsinit.md)
```
Comment (Youssef1313) [inline docs/csharp/whats-new/csharp-9.md:17]:
```suggestion
  - [Suppress emitting localsinit flag](~/_csharplang/proposals/csharp-9.0/skip-localsinit.md)
```

File: docs/csharp/whats-new/csharp-9.md [hunk 4]
```diff
@@ -12,20 +12,20 @@ C# 9.0 adds the following features and enhancements to the C# language:
...
 - [Fit and finish features](#fit-and-finish-features)
-  - Target-typed `new` expressions
-  - static anonymous functions
-  - Target-typed conditional expressions
-  - Covariant return types
-  - Extension `GetEnumerator` support for `foreach` loops
-  - Lambda discard parameters
-  - Attributes on local functions
+  - [Target-typed `new` expressions](../../../../../../csharplang/blob/main/proposals/csharp-9.0/target-typed-new.md)
```
Comment (Youssef1313) [inline docs/csharp/whats-new/csharp-9.md:19]:
```suggestion
  - [Target-typed `new` expressions](~/_csharplang/proposals/csharp-9.0/target-typed-new.md)
```

File: docs/csharp/whats-new/csharp-9.md [hunk 5]
```diff
@@ -12,20 +12,20 @@ C# 9.0 adds the following features and enhancements to the C# language:
...
+  - [`static` anonymous functions](../../../../../../csharplang/blob/main/proposals/csharp-9.0/static-anonymous-functions.md)
```
Comment (Youssef1313) [inline docs/csharp/whats-new/csharp-9.md:20]:
```suggestion
  - [`static` anonymous functions](~/_csharplang/proposals/csharp-9.0/static-anonymous-functions.md)
```

File: docs/csharp/whats-new/csharp-9.md [hunk 6]
```diff
@@ -12,20 +12,20 @@ C# 9.0 adds the following features and enhancements to the C# language:
...
+  - [Target-typed conditional expressions](../../../../../../csharplang/blob/main/proposals/csharp-9.0/target-typed-conditional-expression.md)
```
Comment (Youssef1313) [inline docs/csharp/whats-new/csharp-9.md:21]:
```suggestion
  - [Target-typed conditional expressions](~/_csharplang/proposals/csharp-9.0/target-typed-conditional-expression.md)
```

File: docs/csharp/whats-new/csharp-9.md [hunk 7]
```diff
@@ -12,20 +12,20 @@ C# 9.0 adds the following features and enhancements to the C# language:
...
+  - [Covariant return types](../../../../../../csharplang/blob/main/proposals/csharp-9.0/covariant-returns.md)
```
Comment (Youssef1313) [inline docs/csharp/whats-new/csharp-9.md:22]:
```suggestion
  - [Covariant return types](~/_csharplang/proposals/csharp-9.0/covariant-returns.md)
```

File: docs/csharp/whats-new/csharp-9.md [hunk 8]
```diff
@@ -12,20 +12,20 @@ C# 9.0 adds the following features and enhancements to the C# language:
...
+  - [Extension `GetEnumerator` support for `foreach` loops](../../../../../../csharplang/blob/main/proposals/csharp-9.0/extension-getenumerator.md)
```
Comment (Youssef1313) [inline docs/csharp/whats-new/csharp-9.md:23]:
```suggestion
  - [Extension `GetEnumerator` support for `foreach` loops](../../../../../../csharplang/blob/main/proposals/csharp-9.0/extension-getenumerator.md)
```
```suggestion
  - [Extension `GetEnumerator` support for `foreach` loops](~/_csharplang/proposals/csharp-9.0/extension-getenumerator.md)
```

File: docs/csharp/whats-new/csharp-9.md [hunk 9]
```diff
@@ -12,20 +12,20 @@ C# 9.0 adds the following features and enhancements to the C# language:
...
+  - [Lambda discard parameters](../../../../../../csharplang/blob/main/proposals/csharp-9.0/lambda-discard-parameters.md)
```
Comment (Youssef1313) [inline docs/csharp/whats-new/csharp-9.md:24]:
```suggestion
  - [Lambda discard parameters](~/_csharplang/proposals/csharp-9.0/lambda-discard-parameters.md)
```

File: docs/csharp/whats-new/csharp-9.md [hunk 10]
```diff
@@ -12,20 +12,20 @@ C# 9.0 adds the following features and enhancements to the C# language:
...
+  - [Attributes on local functions](../../../../../../csharplang/blob/main/proposals/csharp-9.0/local-function-attributes.md)
```
Comment (Youssef1313) [inline docs/csharp/whats-new/csharp-9.md:25]:
```suggestion
  - [Attributes on local functions](~/_csharplang/proposals/csharp-9.0/local-function-attributes.md)
```

File: docs/csharp/whats-new/csharp-9.md [hunk 11]
```diff
@@ -12,20 +12,20 @@ C# 9.0 adds the following features and enhancements to the C# language:
...
 - [Support for code generators](#support-for-code-generators)
-  - Module initializers
-  - New features for partial methods
+  - [Module initializers](../../../../../../csharplang/blob/main/proposals/csharp-9.0/module-initializers.md)
```
Comment (Youssef1313) [inline docs/csharp/whats-new/csharp-9.md:27]:
```suggestion
  - [Module initializers](~/_csharplang/proposals/csharp-9.0/module-initializers.md)
```

File: docs/csharp/whats-new/csharp-9.md [hunk 12]
```diff
@@ -12,20 +12,20 @@ C# 9.0 adds the following features and enhancements to the C# language:
...
+  - [New features for partial methods](../../../../../../csharplang/blob/main/proposals/csharp-9.0/extending-partial-methods.md)
```
Comment (Youssef1313) [inline docs/csharp/whats-new/csharp-9.md:28]:
```suggestion
  - [New features for partial methods](~/_csharplang/proposals/csharp-9.0/extending-partial-methods.md)
```

**Review by Youssef1313 — CHANGES_REQUESTED** [review]
This should fix the build warnings. To apply all the suggestions at one commit, go to the "Files changed" tab, and use "Add suggestion to batch" on each suggestion.

File: docs/csharp/whats-new/csharp-9.md (hunk 1 above)
Comment (g-u-r-k-a-n) [reply docs/csharp/whats-new/csharp-9.md:15]: Thanks for your help but I have already tried the tilde character for the root before sent the commit. I mean, your that url -and the others- generates https://github.com/dotnet/docs/blob/main/docs/csharp/whats-new/~/_csharplang/proposals/csharp-9.0/native-integers.md and this link is returning HTTP status code 404 normally. And also you can try `chsarp` instead `_chsarp` path. The result is the same.

Comment (Youssef1313) [reply docs/csharp/whats-new/csharp-9.md:15]: @g-u-r-k-a-n My suggestion will produce 404 when viewing in GitHub, but will work on the actual docs site. Your current approach is the opposite. Using `~/_csharplang` is specific to how building the repository works, and is defined in the configuration file:

https://github.com/dotnet/docs/blob/b5a8fecd1c24be9db5267395e5f6d0c1b514e98f/.openpublishing.publish.config.json#L48-L53

Comment (g-u-r-k-a-n) [reply docs/csharp/whats-new/csharp-9.md:15]: Aww, just started browsing the documents today. I didn't know that config file. Then I will approve your suggestions soon.

**Review by BillWagner — APPROVED** [review]
I like this change @g-u-r-k-a-n 

I'll start formatting the articles like this for C# 10 moving forward.

I've reviewed the changes, and I'll :shipit: now.

**Comment by BillWagner** [comment]
Thanks @Youssef1313 for helping with the build warnings.

**Comment by g-u-r-k-a-n** [comment]
Great idea @BillWagner, it's good to hear that. 👍 

I think more detailed information can be obtained in this way and I like it already. 👌

**Pull request merged**